src/main/java/org/folio/service/budget/BudgetService.java

```diff
@@ -238,40 +200,41 @@ public void clearReadOnlyFields(Budget budgetFromNew) {
    * This method is used exclusively for "Move allocation" action and is skipped for other types.
    *
    * @param transaction The transaction to be checked.
-   * @param client      The database client used for database operations.
+   * @param conn      database connection
    * @return A {@link Future} that completes successfully if there is enough money available,
    * or fails with a {@link HttpException} if not enough money is available.
    */
-  public Future<Void> checkBudgetHaveMoneyForTransaction(Transaction transaction, DBClient client) {
+  public Future<Void> checkBudgetHaveMoneyForTransaction(Transaction transaction, DBConn conn) {
     if (isNotMoveAllocation(transaction)) {
       return Future.succeededFuture();
     }
 
-    return getBudgetByFundIdAndFiscalYearId(transaction.getFiscalYearId(), transaction.getFromFundId(), client, true).compose(budget -> {
-      if (budget.getAvailable() < transaction.getAmount()) {
-        ErrorCodes errorCode = transaction.getTransactionType() == Transaction.TransactionType.ALLOCATION ?
-          NOT_ENOUGH_MONEY_FOR_ALLOCATION : GENERIC_ERROR_CODE;
-        logger.error(errorCode.getDescription());
-        return Future.failedFuture(new HttpException(Response.Status.BAD_REQUEST.getStatusCode(),
-          JsonObject.mapFrom(new Errors().withErrors(singletonList(errorCode.toError())).withTotalRecords(1)).encode()));
-      }
-      return Future.succeededFuture();
-    });
+    return getBudgetByFundIdAndFiscalYearId(transaction.getFiscalYearId(), transaction.getFromFundId(), conn)
+      .compose(budget -> {
+        if (budget.getAvailable() < transaction.getAmount()) {
+          ErrorCodes errorCode = transaction.getTransactionType() == Transaction.TransactionType.ALLOCATION ?
+            NOT_ENOUGH_MONEY_FOR_ALLOCATION : GENERIC_ERROR_CODE;
+          logger.error(errorCode.getDescription());
+          return Future.failedFuture(new HttpException(Response.Status.BAD_REQUEST.getStatusCode(),
```

Comment:
in previous places in the pr we just throw an exception without wrapping into faied future, for consistency we can follow the same approach

Reply:
done